doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,605 @@
+* ``global_identifier`` (integer): The integer identifier of the policy from the global datastore that this configured policy should implement. Required when ``local`` is false, ignored otherwise.
+* ``connections`` (list of ConnectionPoints): The connection points on which this policy applies.
+
+
+Connection Points
+'''''''''''''''''''
+A connection point in the YAML file is defined by providing the ingress and egress of the connection as well as the integer MPLS label that should be applied to enforce the policy on that connection. The egress can be either an interface, an IPv4/IPv6 prefix or a wildcard. The ingress of a connection point is limited to a wildcard or interface.The rationale behind this is that an IPv4/IPv6 ingress would indicate a packet coming from an endhost. Upon arrival at the border router, the packet would have already traversed the internal network and the border router would not be able to enforce a policy (e.g. by applying an MPLS label).
+
+Example of a list of connection points:
+
+::
+
+    - ingress:
+          type: interface
+          interface: 2
+      egress:
+          type: interface
+          interface: 1
+      mpls_label: 1    
+    - ingress:
+          type: interface
+          interface: 2
+      egress:
+          type: ipv4
+          ip: 192.168.5.1
+          prefix: 24
+      mpls_label: 55
+    - ingress:
+          type: wildcard
+      egress:
+          type: interface
+          interface: 2
+      mpls_label: 3
+
+
+
+Border router
+^^^^^^^^^^^^^^^
+
+For a router to query the DRKey secret value from the control service, once has to enable this.
+
+This could look like this::
+
+    [router]
+    use_drkey = true
+
+Considerations for future work
+--------------------------------
+
+SCMP response
+^^^^^^^^^^^^^^^
+
+With the current implementation, the sending endhost is not being informed when his packet gets dropped due to a FABRID error.
+In the future the border routers we might think about whether to enable routers to send an SCMP response if they encounter
+an error when processing FABRID which might help the sending endhost in figuring out why his packet does not arrive at its destination.
+However, we have to be careful to not introduce additional attack vectors, e.g. for volumetric denial-of-service or reflection attacks.
+
+EPIC-HP as extension with Identifier option
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We could create a new HBH extension for EPIC hidden-path, which uses the Identifier option, instead of the
+current design which uses a SCION path-type.
+This allows the use of EPIC HP also in a incremental deployment like we have with FABRID.
+And additionally, we could also use FABRID together with EPIC HP.
+
+.. _RAINBOW:
+
+The RAINBOW system
+^^^^^^^^^^^^^^^^^^^^^
+
+The RAINBOW system allows marking traffic as higher quality of service, to be
+prioritized at on-path BRs.
+RAINBOW can be implemented based on FABRID.
```

Comment:
We have designed FABRID such that RAINBOW can be implemented on top of it.

Reply:
Done